PublicResolver.lll

```diff
@@ -0,0 +1,246 @@
+      code-body))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Hashes a given node to determine its actual storage location.
+  ;; @param node Hash this node address.
+
+  (def 'hash-location (node)
+    (seq
+      (mstore scratch-one node)
+      (sha3 scratch-one 32)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Looks up the address for the node passed in.
+  ;; @param node Get the address for this node.
+
+  (def 'get-address (node)
+    (sload (hash-location node)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Stores the address for the node passed in.
+  ;; @param node Set the address for this node.
+  ;; @param addr The address to set.
+
+  (def 'set-address (node addr)
+    (sstore (hash-location node) addr))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Looks up the content hash for the node passed in.
+  ;; @param node Get the content hash for this node.
+
+  (def 'get-hash (node)
+    (sload (+ (hash-location node) 1)))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Stores the content hash for the node passed in.
+  ;; @param node Set the content hash for this node.
+  ;; @param hash The content hash to set.
+
+  (def 'set-hash (node hash)
+    (sstore (+ (hash-location node) 1) hash))
+
+  ;; --------------------------------------------------------------------------
+  ;; @notice Checks that the caller is the node owner.
+  ;; @param node Check the owner of this node.
+
+  (def 'only-owner (node)
+    (seq
+
+      ;; Call ens.owner(node).
+      (mstore (+ call-data 0x00) (shift-left get-node-owner))
+      (mstore (+ call-data 0x04) node)
+      (call (- (gas) 1000) 0 @@ens-registry call-data 36 return-data 32)
+
+      ;; If the call failed or the caller isn't the current node owner,
+      ;; throw an exception.
+      (when (|| (= @return-code false) (!= @return-data (caller)))
+        (jump invalid-location))))
+
+  ;; --------------------------------------------------------------------------
+  ;; INIT
+
+  ;; Store the constructor parameter.
+  (codecopy scratch-one (bytecodesize) 32)
+  (sstore ens-registry @scratch-one)
+
+  ;; --------------------------------------------------------------------------
+  ;; CODE
+
+  (returnlll
+    (seq
+
+      ;; ----------------------------------------------------------------------
+      ;; @notice Returns true if specified node has specified record type.
+      ;; @param node The ENS node to query.
+      ;; @param kind The record type name, as specified in EIP137.
+      ;; @return True if this resolver has a record of the provided type on the
+      ;;         provided node.
+
+      (def 'node (calldataload 0x04))
+      (def 'kind (calldataload 0x24))
+
+      (function has-record-type
```

Comment:
Given we've deprecated this, I'd just drop it from the lll implementation, personally.